docs/ingress-tls.md

```diff
@@ -0,0 +1,285 @@
+# Using Secrets Store CSI and Azure Key Vault Provider to Enable NGINX Ingress Controller with TLS
+
+This guide demonstrates steps required to setup Secrets Store CSI driver and Azure Key Vault Provider to enable applications to work with NGINX Ingress Controller with TLS certificates stored in Key Vault.
+For more information on securing an Ingress with TLS, refer to [this guide](https://kubernetes.io/docs/concepts/services-networking/ingress/#tls)
+
+Importing the ingress TLS certificate to the cluster can be included in the following deployments:
+
+* Ingress Controller - Ingress deployment is modified to declare and mount the csi provider volume. The secret is imported when Ingress pods are created and the application's pods have no access to the TLS certificate. This scenario fits teams where Ingress and its related certificates are managed centrally by an IT unit.
+* Application - Application deployment manifest declares and mounts the csi provider volume. Only when the application is deployed the certificate is made available in the cluster and when it is removed the secret is gone. This scenario fits development teams who are responsible for the application's security infrastructure and their integration with the cluster.
+
+## Generate a TLS Cert
+
+```bash
+export CERT_NAME=ingresscert
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
+    -out ingress-tls.crt \
+    -keyout ingress-tls.key \
+    -subj "/CN=demo.test.com/O=ingress-tls"
+```
+
+### Import the TLS certificate to Azure Key Vault
+
+Convert the .crt certificate to pfx format and import it to Azure Key Vault. For example:
+
+```bash
+export PASSOWRD="[YOUR PASSWORD]"
+export AKV_NAME="[YOUR AKV NAME]"
+openssl pkcs12 -export -in ingress-tls.crt -inkey ingress-tls.key  -out $CERT_NAME.pfx -passout pass:$PASSWORD
```

Comment:
The AKV provider for CSI driver currently converts PFX to PEM without a password. If the content is password protected then the conversion will fail.

Ref: https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/pkg/provider/provider.go#L434

Reply:
removed password from script.